From patchwork Thu Mar 30 19:29:53 2017
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [U-Boot,2/3] config: am335x_evm: detect Green Wireless using GW1
From: robertcnelson@gmail.com
X-Patchwork-Id: 745390
Message-Id: <20170330192954.22673-2-robertcnelson@gmail.com>
To: u-boot@lists.denx.de
Cc: Tom Rini <trini@konsulko.com>, Jason Kridner <jkridner@beagleboard.org>, 
 Will Newton <willn@resin.io>
Date: Thu, 30 Mar 2017 14:29:53 -0500

SeeedStudio BeagleBone Green Wireless (BBGW) is an expansion of the
SeeedStudio Green (BBG) with the Ethernet replaced by a TI wl1835
wireless module.

This board can be indentified by the GW1x value after A335BNLT (BBB)
in the at24 eeprom:
GW1x [aa 55 33 ee 41 33 33 35  42 4e 4c 54 47 57 31 41  |.U3.A335BNLTGW1A|]

http://beagleboard.org/green-wireless
http://wiki.seeed.cc/BeagleBone_Green_Wireless/

firmware: https://github.com/beagleboard/beaglebone-black-wireless/tree/master/firmware
wl18xx mac address: Stored in at24 eeprom at address 5-16:
hexdump -e '8/1 "%c"' /sys/bus/i2c/devices/0-0050/eeprom | cut -b 5-16

Upstream-Status: Pending

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
CC: Tom Rini <trini@konsulko.com>
CC: Jason Kridner <jkridner@beagleboard.org>
CC: Will Newton <willn@resin.io>
---
 board/ti/am335x/board.c      | 4 ++++
 include/configs/am335x_evm.h | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/board/ti/am335x/board.c b/board/ti/am335x/board.c
index 304ae34fb7..54589638b3 100644
--- a/board/ti/am335x/board.c
+++ b/board/ti/am335x/board.c
@@ -656,6 +656,10 @@ int board_late_init(void)
 		if (!strncmp(board_ti_get_rev(), "BWA", 3)) {
 			name = "BBBW";
 		}
+		/* SeeedStudio BeagleBone Green Wireless */
+		if (!strncmp(board_ti_get_rev(), "GW1", 3)) {
+			name = "BBGW";
+		}
 	}
 
 	if (board_is_bbg1())
diff --git a/include/configs/am335x_evm.h b/include/configs/am335x_evm.h
index dcac2f5816..d90c5156d3 100644
--- a/include/configs/am335x_evm.h
+++ b/include/configs/am335x_evm.h
@@ -142,6 +142,8 @@
 			"setenv fdtfile am335x-boneblack-wireless.dtb; fi; " \
 		"if test $board_name = BBG1; then " \
 			"setenv fdtfile am335x-bonegreen.dtb; fi; " \
+		"if test $board_name = BBGW; then " \
+			"setenv fdtfile am335x-bonegreen-wireless.dtb; fi; " \
 		"if test $board_name = A33515BB; then " \
 			"setenv fdtfile am335x-evm.dtb; fi; " \
 		"if test $board_name = A335X_SK; then " \
